composer/workflows/bq_copy_across_locations.py

```diff
@@ -0,0 +1,194 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example Airflow DAG that performs an export from BQ tables listed in
+config file to GCS,
+copies GCS objects across locations (e.g., from US to EU) then imports from
+GCS to BQ. The DAG
+imports the gcs_to_gcs operator from plugins and dynamically builds the
+tasks based on the
+list of tables.
+Lastly, the DAG defines a specific application logger to generate logs.
+
+This DAG relies on three Airflow variables
+https://airflow.apache.org/concepts.html#variables
+* master_file_path - CSV file listing source and target tables, including
+Datasets.
+* gcs_source_bucket - Google Cloud Storage bucket to use for exporting
+BigQuery tables in source.
+* gcs_dest_bucket - Google Cloud Storage bucket to use for importing
+BigQuery tables in
+                    destination.
+  See https://cloud.google.com/storage/docs/creating-buckets for creating a
+  bucket.
+"""
+
+# --------------------------------------------------------------------------------
+# Load The Dependencies
+# --------------------------------------------------------------------------------
+
+import csv
+from datetime import datetime
+from datetime import timedelta
+import logging
+
+from airflow import models
+from airflow.contrib.operators.bigquery_to_gcs import \
+    BigQueryToCloudStorageOperator
+from airflow.contrib.operators.gcs_to_bq import \
+    GoogleCloudStorageToBigQueryOperator
+from airflow.models import Variable
+from airflow.operators.dummy_operator import DummyOperator
+# Import operator from plugins
+from gcs_plugin.operators.gcs_to_gcs import \
+    GoogleCloudStorageToGoogleCloudStorageOperator
+import google.cloud.logging
+
+
+# --------------------------------------------------------------------------------
+# Set default arguments
+# --------------------------------------------------------------------------------
+
+default_args = {
+    'owner': 'airflow',
+    'start_date': datetime.today(),
+    'depends_on_past': False,
+    'email': [''],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': timedelta(minutes=5),
+}
+
+# --------------------------------------------------------------------------------
+# Set variables
+# --------------------------------------------------------------------------------
+
+# 'master_file_path': This variable will contain the location of the master
+# file.
+master_file_path = Variable.get('master_file_path')
+
+# Source Bucket
+source_bucket = Variable.get('gcs_source_bucket')
+
+# Destination Bucket
+dest_bucket = Variable.get('gcs_dest_bucket')
+
+# --------------------------------------------------------------------------------
+# Set GCP logging
+# --------------------------------------------------------------------------------
+
+# Instantiates a client
+client = google.cloud.logging.Client()
```

Comment:
Is it common to set up logging within DAG parsing? Won't Airflow and/or Composer ensure logs get written to the correct place? Why is this code present?

Reply:
Code is actually there to test logging against GCP Stackdriver while running in local Airflow environment, I would remove to keep the example clean as it is less relevant with Composer sending logs automatically to Stackdriver.